Sources/FluentMongo/MongoConnection.swift

```diff
@@ -45,74 +50,92 @@ public final class MongoConnection: BasicWorker, DatabaseConnection, DatabaseQue
 
     /// Closes the `DatabaseConnection`.
     public func close() {
-        self.client.close()
         self.isClosed = true
     }
 
     /// See `DatabaseQueryable`.
 
     public func query(_ query: Database.Query, _ handler: @escaping (Database.Output) throws -> Void) -> Future<Void> {
-        do {
+
+        let promise = self.eventLoop.newPromise(of: Void.self)
+        self.threadPool.submit { _ in
```

Comment:
@tanner0101 - can I get your opinion on the usage of `BlockingIOThreadPool` throughout `MongoConnection.swift`?
In particular I am not sure in which instance I should return to the `eventLoop`.